ui/src/message_popup/hint_card.jsx

```diff
@@ -0,0 +1,141 @@
+// @flow
+import React from 'react';
+import * as PropTypes from '../prop_types.js';
+import RaisedButton from 'material-ui/RaisedButton';
+import Divider from 'material-ui/Divider';
+
+/*
+Shows a card for a student from the virtual school.
+*/
+export default React.createClass({
+  displayName: 'HintCard',
+
+  getInitialState: function(){
+    return ({
+      hidden: true,
+      exampleIndex: 0,
+      nonExampleIndex: 0,
+    });
+  },
+  
+  propTypes: {
+    examples: React.PropTypes.array.isRequired,
+    nonExamples: React.PropTypes.array.isRequired
+  },
+  
+  onNextExample(){
+    var length = this.props.examples.length;
+    var next = this.state.exampleIndex + 1;
+    if(next >= length){
+      next = 0;
+    }
+    this.setState({exampleIndex: next});
+  },
+  
+  onNextNonExample(){
+    var length = this.props.nonExamples.length;
+    var next = this.state.nonExampleIndex + 1;
+    if(next >= length){
+      next = 0;
+    }
+    this.setState({nonExampleIndex: next});
+  },
+  
+  onHintsToggled(){
+    this.setState({ hidden: !this.state.hidden });
+  },
+
+  render() {
```

Comment:
I think this also could help simplify the layout of the component here in code too let me know what you think.